UNFCCC_GHG_data/UNFCCC_reader/Guinea/read_GIN_BUR1_from_pdf.py

```diff
@@ -0,0 +1,905 @@
+                                tables_inventory_original[1].df[3 :-2]],
+                               axis=0,
+                               join='outer').reset_index(drop=True)
+
+    row_to_delete = df_energy_year.index[
+        df_energy_year[0] == '1.A.3.a.i - Aviation internationale (Soutes internationales)'][0]
+    df_energy_year = df_energy_year.drop(index=row_to_delete)
+
+    row_to_delete = df_energy_year.index[
+        df_energy_year[0] == '1.A.3.d.i - Navigation internationale (soutes internationales)'][0]
+    df_energy_year = df_energy_year.drop(index=row_to_delete)
+
+    row_to_delete = df_energy_year.index[
+        df_energy_year[0] == '1.A.5.c - Opérations multilatérales (Éléments pour information)'][0]
+    df_energy_year = df_energy_year.drop(index=row_to_delete)
+
+    # add header and unit
+    df_header = pd.DataFrame([inv_conf["header_energy"], inv_conf["unit_energy"]])
+
+    df_energy_year = pd.concat([df_header, df_energy_year], axis=0, join='outer').reset_index(drop=True)
+
+    df_energy_year = pm2.pm2io.nir_add_unit_information(df_energy_year,
+                                                        unit_row=inv_conf["unit_row"],
+                                                        entity_row=inv_conf["entity_row"],
+                                                        regexp_entity=".*",
+                                                        regexp_unit=".*",
+                                                        default_unit="Gg")
+
+    print("Added unit information.")
+    # set index
+    df_energy_year = df_energy_year.set_index(inv_conf["index_cols"])
+
+    # convert to long format
+    df_energy_year_long = pm2.pm2io.nir_convert_df_to_long(df_energy_year, inv_conf["year"][page],
+                                                           inv_conf["header_long"])
+
+    # extract from tuple
+    df_energy_year_long["orig_cat_name"] = df_energy_year_long["orig_cat_name"].str[0]
+
+    # prep for conversion to PM2 IF and native format
+    # make a copy of the categories row
+    df_energy_year_long["category"] = df_energy_year_long["orig_cat_name"]
+
+    # replace cat names by codes in col "category"
+    # first the manual replacements
+    df_energy_year_long["category"] = df_energy_year_long["category"].str.replace('\n', '')
+    df_energy_year_long["category"] = \
+        df_energy_year_long["category"].replace(inv_conf["cat_codes_manual"]['energy'])
+
+    df_energy_year_long["category"] = df_energy_year_long["category"].str.replace(".", "")
+
+    # then the regex replacements
+    repl = lambda m : m.group('code')
+    df_energy_year_long["category"] = \
+        df_energy_year_long["category"].str.replace(inv_conf["cat_code_regexp"], repl,
+                                                    regex=True)
+
+    df_energy_year_long = df_energy_year_long.reset_index(drop=True)
+
+    df_energy_year_long["data"] = df_energy_year_long["data"].str.replace(",", ".")
```

Comment:
If these data cleaning operations repeat for each table group maybe put them in a small function to shorten the code

Reply:
OK. I probably repeat them all the time in my scripts as well